### PR TITLE
Fix division by zero in yield calibration

### PR DESCRIPTION
**Pull request by DavidhoPIK**
## :bird: Description of this PR :bird:

This PR fixes a division by zero error that occures when using the h16 or h16s5 region mapping using 67k preprocessing. 

The value ```sum((cell(i,j),w), fm_croparea("y1995",j,w,knbe14))``` in LN 130 of the [preloop](https://github.com/magpiemodel/magpie/blob/master/modules/14_yields/managementcalib_aug19/preloop.gms) is above but very close to zero. Like this the >0 condition is fulfilled but still LN 134 results in division by zero. 

## :wrench: Checklist for PR creator :wrench:

- [x] Label pull request [from the label list](https://github.com/magpiemodel/magpie/labels).
  - **Low risk**: Simple bugfixes (missing files, updated documentation, typos) or changes  in start or output scripts
  - **Medium risk**: Uncritical changes in the model core (e.g. moderate modifications in non-default realizations)
  - **High risk**: Critical changes in model core or default settings (e.g. changing a model default or adjusting a core mechanic in the model)

- [x] Self-review own code
  - No hard coded numbers and cluster/country/region names.
  - The new code doesn't contain declared but unused parameters or variables.
  - [`magpie4`](https://github.com/pik-piam/magpie4) R library has been updated accordingly and backwards compatible where necessary.
  - `scenario_config.csv` has been updated accordingly (important if `default.cfg` has been updated)

- [x] Document changes 
  - Add changes to `CHANGELOG.md`
  - Where relevant, put In-code documentation comments
  - Properly address updates in interfaces in the module documentations
  - run [`goxygen::goxygen()`](https://github.com/pik-piam/goxygen) and verify the modified code is properly documented

- [x] Perform test runs
  - **Low risk**: 
    - Run a compilation check via `Rscript start.R --> "compilation check"`
  - **Medium risk**: 
    - Run test runs via `Rscript start.R --> "test runs"`
    - Check logs for errors/warnings
  - **High risk**:
    - Run test runs via `Rscript start.R --> "test runs"`
    - Check logs for errors/warnings
    - Default run from the PR target branch for comparison
    - Provide relevant comparison plots (land-use, emissions, food prices, land-use intensity,...)

### :chart_with_downwards_trend: Performance changes :chart_with_upwards_trend:
  
  - Current develop branch default : ** mins
  - This PR's default :  ** mins

## :rotating_light: Checklist for reviewer :rotating_light:

- PR is labeled correctly
- Code changes look reasonable
  - No hard coded numbers and cluster/country/region names.
  - No unnecessary increase in module interfaces
  - model behavior/performance is satisfactory.
- Changes are properly documented
  - `CHANGELOG` is updated correctly
  - Updates in interfaces have been properly addressed in the module documentations
  - In-code documentation looks appropriate
- [ ] content review done (at least 1)
- [x] RSE review done (at least 1)

### CODE REVIEW

**Review by tscheypidi — APPROVED** [review]
can you move the changelog entry from "changed" to "fixed" as this more seems to be a bugfix? Other than that it looks good

File: modules/14_yields/managementcalib_aug19/preloop.gms [hunk 1]
```diff
@@ -127,9 +127,10 @@ if ((s14_calib_ir2rf = 1),
 * Calibrate newly calibrated yields to FAO yields
   i14_modeled_yields_hist2(i,knbe14)
    = (sum((cell(i,j),w), fm_croparea("y1995",j,w,knbe14) * i14_yields_calib("y1995",j,knbe14,w)) /
-      sum((cell(i,j),w), fm_croparea("y1995",j,w,knbe14)))$(sum((cell(i,j),w), fm_croparea("y1995",j,w,knbe14))>0)
+      sum((cell(i,j),w), fm_croparea("y1995",j,w,knbe14)))$(sum((cell(i,j),w), fm_croparea("y1995",j,w,knbe14))>0.00001)
    + (sum((cell(i,j),w), i14_croparea_total("y1995",w,j) * f14_yields("y1995",j,knbe14,w)) /
-      sum((cell(i,j),w), i14_croparea_total("y1995",w,j)))$(sum((cell(i,j),w), fm_croparea("y1995",j,w,knbe14))=0);
+      sum((cell(i,j),w), i14_croparea_total("y1995",w,j)))
```
Comment (k4rst3ns) [inline modules/14_yields/managementcalib_aug19/preloop.gms:132]: Please introduce same changes at LN 60 for consistency.

Please check with rounding accuracy in fullCELLULARMAGPIE.

See also: https://github.com/magpiemodel/magpie/blob/580b92202badf95b1708d40c4fb37370f01482d9/modules/30_crop/endo_apr21/preloop.gms#L28

There fm_croparea is already corrected to not have negative values. 
But you can keep the "and [...] >=0", it should not change anything.

**Review by k4rst3ns — CHANGES_REQUESTED** [review]

File: modules/14_yields/managementcalib_aug19/preloop.gms (hunk 1 above)
Comment (DavidhoPIK) [reply modules/14_yields/managementcalib_aug19/preloop.gms]: Thank you for the comment. I added the analogous change as requested and removed the >= check for better readability. 
The rounding of croparea in fullCELLULARMAGPIE is 5. That means adding or subtracting the number ```0.00001``` does make a difference for the croparea.
However the value ```0.000001``` for the check still results in a division by zero.

Comment (DavidhoPIK) [reply modules/14_yields/managementcalib_aug19/preloop.gms]: Now that I thought more about it, I think such a check that uses a number that would be rounded to zero by fullCELLULARMAGPIE could never make any difference. We check for a number that is a sum of positive values that can only be larger than ```0.000001```. It is clear that this number is >= ```0.000001```.

Comment (k4rst3ns) [reply modules/14_yields/managementcalib_aug19/preloop.gms]: Good point.

I thing 10^-5 is a very small number and  an "smaller as" should only be in very few edge cases.
SO lets got with this solution, evenso it is kind of strange, that gams is so inconsistent here.

**Review by k4rst3ns — APPROVED** [review]

File: modules/14_yields/managementcalib_aug19/preloop.gms (hunk 1 above)
Comment (DavidhoPIK) [reply modules/14_yields/managementcalib_aug19/preloop.gms]: Ok, I agree that 10^-5 is a very small number and this change should not result in content-wise differences. Thank you for the review.

**Pull request merged**